### DISCUSSION TITLE
Explorer Plugin Settings

### DISCUSSION THREAD

**Discussion by PCO62** [Flow-Launcher/Flow.Launcher#1993] [Q&A]
Hi

I'm a new Flow Launcher user (and it's my 1st intervention on this forum) and comparing to other launchers (I've tried several) I think it's the one with the most features, great customization capabilities and more potential to become the "best launcher"...
I love it. 
**_Congratulations to the developers! :)_**

As a new user, I have some basic questions on how to use it and how to configure it.
To help clarify my doubts, I have attached an image below signaling the questions with "?".

**Questions related to "Explorer Plugin":**

1. In the tab "General Settings", what is the function of the fields marked with a "?", considering that I have chosen the "Windows Index" as a search source?
2. In the tab "Everything Settings" I have the following questions, taking into account that I have the Everything App with the respective service active:

> i) The marked field will be to insert the Path of the executable (?) of App Everything, or is it, already, integrated in FL (?) and, in the latter case, then what is the respective field for?
> ii) Taking into account the above point, what will be the behavior during the search process with FL?
>     - it "chooses" the "Windows Index" or the "Everything"?
>     - don't they conflict with both being selected?
>     - is there any way to switch between one and the other?

![FL_ExplPlg](https://user-images.githubusercontent.com/93612520/226187983-207a89f1-0dd8-4455-971f-e7d4dfc87230.png)

**_Sorry for these basic questions.
Thanks
...and congratulations once again! :)_**

**Comment by VictoriousRaptor** [comment]
1. For working directory please google it. Basically it's a folder where a program gets some context data.
2. File Editor and Folder Editor is for something like VS Code. Exploerer plugin provice Leave a blank if you don't need it.
3. [Shell](https://en.wikipedia.org/wiki/Shell_(computing))
4. Using Everything or not is determined by the options in General tab.

**Reply by oeloo** [reply]
> 4\. Using Everything or not is determined by the options in General tab.

Where is this option in General tab ? I do not see it:
![image](https://user-images.githubusercontent.com/98819304/227465471-e33b211d-742e-4a85-aff1-721014bdd8dc.png)

**Reply by VictoriousRaptor** [reply]
Oh my bad. I mean "General Settings" tab in Explorer plugin :(

**Reply by oeloo** [reply]
I understand now

**Reply by PCO62** [reply]
> 1. For working directory please google it. Basically it's a folder where a program gets some context data.
> 2. File Editor and Folder Editor is for something like VS Code. Exploerer plugin provice Leave a blank if you don't need it.
> 3. [Shell](https://en.wikipedia.org/wiki/Shell_(computing))
> 4. Using Everything or not is determined by the options in General tab.

_**Hi @VictoriousRapto**_

Thank you in advance for your kindness in responding to my post (many times, experts in some knowledge do not have patience for "newbies" in this knowledge) :)

I think I understood your first two answers:

> 1.  "...working directory of executable"

it will be the source of some resources of the executable that I get in the results panel when I search for it.. is that right?

> 2. "File Editor and Folder Editor"

it will be the Paths for ,e.g., txt editors like notepad++,... and more, isn't it?

> 3.  Shell:

I know that "cmd" corresponds to the executable (cmd.exe) - command prompt - of the Windows OS.
_**My question is when/how to use it and in what context, in FL.**_..

>4.  Using Everything or not is determined by the options in General tab..

"Yes, but"..., (as shown in the figure I attached, which corresponds to the "General Settings" tab of the plugin explorer). 
 My question have more **_particularities/specificities as described in the items "i) and ii)" of point "2."_** in the post. 
... my doubts remain! 



Although it remains some doubts, **_I thank you, again, for your availability, @VictoriousRapto :)_**

**Reply by VictoriousRaptor** [reply]
> I think I understood your first two answers:
> 
> > 1. "...working directory of executable"
> 
> it will be the source of some resources of the executable that I get in the results panel when I search for it.. is that right?
> 
> > 2. "File Editor and Folder Editor"
> 
> it will be the Paths for ,e.g., txt editors like notepad++,... and more, isn't it?
> 

That's right.

> > 3. Shell:
> 
> I know that "cmd" corresponds to the executable (cmd.exe) - command prompt - of the Windows OS. _**My question is when/how to use it and in what context, in FL.**_..

For context menu in explorer plugin. Press Shift+Enter or right click on results provided by explorer plugin and you'll see what I mean. File editor works similarily but you'll need to choose an editor first.

> 
> > 4. Using Everything or not is determined by the options in General tab..
> 
> "Yes, but"..., (as shown in the figure I attached, which corresponds to the "General Settings" tab of the plugin explorer). My question have more **_particularities/specificities as described in the items "i) and ii)" of point "2."_** in the post. ... my doubts remain!
> 

Sorry I didn't see your question clearly in the answer before. Everything is not intergrated in Flow. You can  mananully download it from their offical site. Filling the Everything path is not necessary if Everything is installed and running.

Everything and windows index won't conflict with each other. If you have Everything installed I recommend you to use it for all the options. It's way faster than windows index.

We just updated the [usage](https://www.flowlauncher.com/docs/#/plugin-explorer-usage) of explorer plugin. Hope it will help you.

**Reply by PCO62** [reply]
> We just updated the [usage](https://www.flowlauncher.com/docs/#/plugin-explorer-usage) of explorer plugin. Hope it will help you.

**_Great :)
Thank you very much, @VictoriousRaptor_**

Before posting my 1st post that focused on this subject (the Explorer Plugin is one of the fundamental components of FL), I had looked for information in the FL documentation and "only" found the "Usage Tips" - the other topics are, mostly, of technical nature about how to develop plugins (I'm not at that level!...) - and searching the Discussion Forum, also, I hadn't found the information I needed.

I will come back, with more time, _to answer you_, and after having "studyed" the new information... ; )
**_Thank you_**

**Comment by oeloo** [comment]
> I'm a new Flow Launcher user (and it's my 1st intervention on this forum) and comparing to other launchers (I've tried several) I think it's the one with the most features, great customization capabilities and more potential to become the "best launcher"...

Did you tried these 2 launchers:
- Listary: https://www.listary.com/
- Quick Access Pop-up (QAP): https://www.quickaccesspopup.com/

Does their search result open itself in your current File Explorer windows (or as tab) or in a new File Explorer window?

**Reply by PCO62** [reply]
> > I'm a new Flow Launcher user (and it's my 1st intervention on this forum) and comparing to other launchers (I've tried several) I think it's the one with the most features, great customization capabilities and more potential to become the "best launcher"...
> 
> Did you tried these 2 launchers:
> 
> * Listary: https://www.listary.com/
> * Quick Access Pop-up (QAP): https://www.quickaccesspopup.com/
> 
> Does their search result open itself in your current File Explorer windows (or as tab) or in a new File Explorer window?

Hi @oeloo :)

I've already tested several launchers, including Listary (I confess I don't know the QAP), but as they never satisfied me with their functional characteristics and, mainly, lack of customization of the interface (contrary to what happens with FL), I didn't use them and, for that reason, I stopped using them a long time ago.
Because of that I can't, honestly, answer you directly at the moment... but I'll try to find out (trying again, for example, the "old man" Listary... I don't even know much it evolved...)

Cheers :)

**Reply by PCO62** [reply]
>"Does their search result open itself in your current File Explorer windows (or as tab) or in a new File Explorer window?"

>" ... Because of that I can't, honestly, answer you directly at the moment... but I'll try to find out..."

Thinking better, I suppose not, @oeloo!

**IN THIS CONTEXT:**
From what I've been reading about "launchers", FL is currently considered **_one of the most active in its development_** (if not the most active) and **_with the greatest potential for evolution in its functionalities._**

In addition to its features, I love it because it is **very flexible** and **very customizable** (we can choose/change almost "everything" to customize). 
For example, without any knowledge of programming, I have, already, my own FL themes  - I like the information, aka, results, more condensed, with font sizes between 10 and 12, and with that, I acquire more information in the same space. None of the launchers I tested had this flexibility. 

For the reasons mentioned, I intend to try to get to know it better and more deeply in its capabilities than doing what I've done before of going around testing several launchers without ever knowing them in depth  (... also, it's true, I didn't like its interfaces at all and the absence, or little, possibility of customization).

_It´s because I think I found my ultimate launcher: FL!  ; )_
